components/cdc/src/service.rs

```diff
@@ -155,31 +156,33 @@ impl ChangeData for Service {
 
         let rx = BatchReceiver::new(rx, CDC_MSG_MAX_BATCH_SIZE, Vec::new, VecCollector);
         let send_resp = sink.send_all(
-            rx.map(|events| {
-                // The size of the response should not exceed CDC_MAX_RESP_SIZE.
-                // Split the events into multiple responses by CDC_MAX_RESP_SIZE here.
-                let events_len = events.len();
-                let mut event_vecs = vec![Vec::with_capacity(events_len)];
-                let mut current_events_size = 0;
-                for (size, event) in events {
-                    if current_events_size + size >= CDC_MAX_RESP_SIZE {
-                        event_vecs.push(Vec::with_capacity(events_len));
-                        current_events_size = 0;
+            rx.map(|events| std::result::Result::<_, ()>::Ok(events))
+                .compat()
```

Comment:
How about `compat` at last so that you can use async closure to get rid of map calls.?